### PR TITLE
Feature/pagination sorting swagger

### PR DESCRIPTION
**Pull request by bodya4243**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -16,36 +20,41 @@
...
 @RestController
 @RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
-    
+
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "get all Book", description = "get all the books")
```
Comment (nklimovych) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]:
```suggestion
    @Operation(summary = "Get all Books", description = "Get all the books")
```
Same issue in other `@Operation's`

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -16,36 +20,41 @@
...
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
-    
+
     @GetMapping("/{id}")
     public BookDto getById(@PathVariable Long id) {
-        return bookService.findById(id);
+        return bookService.getById(id);
     }
```
Comment (nklimovych) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]: I suggest adding `@ResponseStatus` to all methods for consistency.

File: src/main/java/com/example/onlinebookstore/dto/BookRequestDto.java [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+    @NotNull
+    @Size(min = 2)
     private String author;
 
+    @NotNull
+    @Min(value = 0, message = "can't be less than 0")
     private BigDecimal price;
 
+    @NotNull
+    @Pattern(regexp = "\\d{13}", message = "ISBN must be a 13-digit number")
```
Comment (nklimovych) [inline src/main/java/com/example/onlinebookstore/dto/BookRequestDto.java]: There's annotation `@ISBN`, so you can use it with `@NotNull` to validate ISBN.
Just a suggestion 🙂

**Review by nklimovych — CHANGES_REQUESTED** [review]
Looks good. Left you some minor comments)

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -16,36 +20,43 @@
...
     @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @PutMapping("/{id}")
-    public BookDto update(@PathVariable Long id, @RequestBody BookRequestDto bookRequestDto) {
+    @ResponseStatus(HttpStatus.OK)
+    public BookDto update(@PathVariable Long id,
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]: Is it possible to add documentation here as well

File: src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: These changes not related to this PR

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Please make only necessary changes in PR

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**